src/libs/resizable_panel.js

```diff
@@ -0,0 +1,170 @@
+const sizes = {
+  0: window.innerHeight,
+  [window.innerHeight / 2]: window.innerHeight / 2,
+  [window.innerHeight]: 50,
+};
+
+const REDUCED = 'reduced';
+const FULL = 'full';
+const resizerClassName = 'deployable__list_resizer';
+export const resizableClassName = 'resizable_panel';
+
+function getClosest(arr, goal) {
+  return arr.reduce((prev, curr) =>
+    Math.abs(curr - goal) < Math.abs(prev - goal) ? curr : prev
+  );
+}
+
+export default class ResizablePanel {
+  constructor() {
+    this.timer = null;
+    this.panel = null;
+    this.holding = false;
+    this.isTransitioning = false;
+
+    this.checkPanel = this.checkPanel.bind(this);
+    this.holdResizer = this.holdResizer.bind(this);
+    this.move = this.move.bind(this);
+    this.stopResize = this.stopResize.bind(this);
+    this.playTransition = this.playTransition.bind(this);
+
+    window.deployedListHold = this.holdResizer;
+    window.deployedListStop = this.stopResize;
+  }
+
+  /**
+   * Looks for a resizable element
+   */
+  checkPanel() {
+    if (!this.panel) {
+      const element = document.querySelector(`.${resizableClassName}`);
+      if (element) {
+        this.panel = element;
+        return true;
+      }
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Triggered on mouse down of the panel resizer
+   * @param {MouseEvent|TouchEvent} e event
+   */
+  holdResizer(e) {
+    e.preventDefault();
+
+    if (!this.checkPanel() || this.isTransitioning) {
+      return;
+    }
+
+    this.move(e, true);
+    this.panel.classList.remove(REDUCED, FULL);
+    fire('panel_view_state');
+
+    this.timer = setTimeout(() => {
+      if (this.timer && this.checkPanel()) {
+        this.holding = true;
+        document.addEventListener('touchmove', this.move);
+        document.addEventListener('mousemove', this.move);
+      }
+    }, 250);
+  }
+
+  /**
+   * Triggered on mouse move inside the body element while holding the panel resizer
+   * @param {MouseEvent|TouchEvent} e event
+   * @param {boolean} force Apply the height outside a panel resize
+   */
+  move(e, force = false) {
+    e.preventDefault();
+
+    const resizer = document.querySelector(`.${resizerClassName}`);
+
+    if (!resizer) {
+      return;
+    }
+
+    if ((this.holding || force) && this.checkPanel() && !this.isTransitioning) {
+      const clientY = e.changedTouches ? e.touches[0].clientY : e.clientY;
```

Comment:
You should check if `e.touches` exists before trying to access one of its values.